### PR TITLE
Add support for typeorm ^0.3.0

### PR DESCRIPTION
**Pull request by dominicbachmann**
I had to change quite a lot since the way of defining custom repositories has heavily changed. All the tests are passing. I've added one change that is not related to the typeorm upgrade because I tought that it is probably a bug (the change is marked with: // TODO: Not sure about this change (key was just id before).

Please let me know if this is a valid approach and if I can do anything else.

### CODE REVIEW

**Comment by erishiraj** [comment]
This merge will resolve my issue as well.

**Comment by rpvsilva** [comment]
Hi, nice work here. @bashleigh is it possible to merge this and publish a new version?

File: src/__tests__/polymorphic.repository.spec.ts [hunk 1]
```diff
@@ -103,10 +113,13 @@ describe('AbstractPolymorphicRepository', () => {
         });
       });
     });
-
+    //
```
Comment (bashleigh) [inline src/__tests__/polymorphic.repository.spec.ts:116]: need this comment line?

File: src/__tests__/polymorphic.repository.spec.ts [hunk 2]
```diff
@@ -117,18 +130,25 @@ describe('AbstractPolymorphicRepository', () => {
...
+        });
 
         expect(result).toBeInstanceOf(AdvertEntity);
         expect(result.owner).toBeInstanceOf(UserEntity);
         expect(result.owner.id).toBe(result.entityId);
         expect(result.entityType).toBe(UserEntity.name);
       });
     });
-
+    //
```
Comment (bashleigh) [inline src/__tests__/polymorphic.repository.spec.ts:145]: And this one?

**Review by bashleigh — APPROVED** [review]
Looks good, couldn't find your TODO/bug that you mentioned? Have requested the empty comment lines to be removed, thanks :)

File: src/polymorphic.repository.ts [hunk 3]
```diff
@@ -34,378 +32,306 @@ const entityIdColumn = (options: PolymorphicMetadataInterface): string =>
...
+    ): Promise<
+      PolymorphicChildInterface[] | PolymorphicChildInterface | never
+    > {
+      const repository = this.findRepository(entityType);
+
+      return repository[options.hasMany ? 'find' : 'findOne'](
+        options.type === 'parent'
+          ? {
+              where: {
+                // TODO: Not sure about this change (key was just id before)
```
Comment (rpvsilva) [inline src/polymorphic.repository.ts:170]: @bashleigh He's talking about this

Comment (rpvsilva) [reply src/polymorphic.repository.ts:170]: Also, to fix the tests please add a volume to MySQL service, it will work

**Comment by sashalikesplanes** [comment]
Anything I can help with @bashleigh @dominicbachmann ? Would be awesome to finish the bump to typeorm ^0.3.0

**Comment by bashleigh** [comment]
@sashalikesplanes Yes please, could you fix your tests? Looks like they're not passing in github actions. Looks like config setup issues

**Comment by rpvsilva** [comment]
@dominicbachmann You just need to rebase with the master branch and it will be fixed

**Comment by dominicbachmann** [comment]
> @dominicbachmann You just need to rebase with the master branch and it will be fixed

Will do

**Comment by tiagomichaelsousa** [comment]
Any further updates regarding this topic? 🚀

**Comment by lowbits** [comment]
Hey nice work, im curious i forked your branch and was trying the inverse of the provided example.

```ts
@Entity()
export class StoreEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @PolymorphicChildren(() => [VisitorEntity, EmployeeEntity], {
        entityTypeId: 'id'
    })
    visitors: (VisitorEntity | EmployeeEntity)[];
}

@Entity()
export class VisitorEntity{
  @PrimaryGeneratedColumn()
  id: number;

  @PolymorphicParent(() => StoreEntity, {
    eager: true,
  })
  store?: StoreEntity;


  @Column()
  entityType: string;
}


@Entity()
export class EmployeeEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  entityType: string;

  @PolymorphicParent(() => StoreEntity, {
    eager: true,
  })
  store: StoreEntity;
}
```

But i think the provided package doesnt work with this example?
Do you have any hints how to make it work?


Best 

Tobi

**Comment by jspizziri** [comment]
After looking more closely at this PR, I've decided to submit a different approach to this upgrade.

Please see https://github.com/bashleigh/typeorm-polymorphic/pull/38 which explains the motivation.

**Pull request closed without merging**